src/Resources/ResourceManager/Implementation/Deployments/GetAzureDeploymentOperationCmdlet.cs

```diff
@@ -14,42 +14,73 @@
     using System.Management.Automation;
-    using System.Threading.Tasks;
-    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
     using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
     using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
-    using Microsoft.Azure.Commands.ResourceManager.Common;
-    using Newtonsoft.Json.Linq;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels.Deployments;
+    using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
 
     /// <summary>
     /// Gets the deployment operation.
     /// </summary>
-    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeploymentOperation", DefaultParameterSetName = GetAzureDeploymentOperationCmdlet.DeploymentNameParameterSet), OutputType(typeof(PSDeploymentOperation))]
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeploymentOperation", DefaultParameterSetName = GetAzureDeploymentOperationCmdlet.SubscriptionParameterSetWithDeploymentName), OutputType(typeof(PSDeploymentOperation))]
     public class GetAzureDeploymentOperationCmdlet : ResourceManagerCmdletBase
     {
-        /// <summary>
-        /// The deployment name parameter set.
-        /// </summary>
-        internal const string DeploymentNameParameterSet = "GetByDeploymentName";
+        internal const string ResourceGroupParameterSetWithDeploymentName = "ResourceGroupWithDeploymentName";
+        internal const string SubscriptionParameterSetWithDeploymentName = "SubscriptionWithDeploymentName";
+        internal const string ManagementGroupParameterSetWithDeploymentName = "ManagementGroupWithDeploymentName";
+        internal const string TenantParameterSetWithDeploymentName = "TenantWithDeploymentName";
 
-        /// <summary>
-        /// The deployment object parameter set.
-        /// </summary>
         internal const string DeploymentObjectParameterSet = "GetByDeploymentObject";
 
+        [Parameter(ParameterSetName = GetAzureDeploymentOperationCmdlet.ResourceGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(ParameterSetName = GetAzureDeploymentOperationCmdlet.SubscriptionParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(ParameterSetName = GetAzureDeploymentOperationCmdlet.ManagementGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(ParameterSetName = GetAzureDeploymentOperationCmdlet.TenantParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public DeploymentScopeType? ScopeType { get; set; }
```

Comment:
Switch parameter would only be able to indicate true/false. originally we wanted that as we planed to use to tell whether it's subscription or tenant scope. In our subsequent discussions we decided to do a breaking change and release it in a major release, and we decided to just make it consistent in all four scopes, by using this ScopeType parameter.

Reply:
The problem is that this doesn't allow users to fall into the pit of success with the parameters - they can select a scope type and provide parameters for a non-matching scope, resulting in a runtime error.  If you use switch parameters, the user falls into the pit of success - they can only provide parameters for the deployment type they have selected.

Since this is a preview, I won't prevent you from doing this, but you will need to match up with the origiinal design review once this goes GA.